cincinnati/src/plugins/internal/arch_filter.rs

```diff
@@ -0,0 +1,271 @@
+                (None, Some(version)) => infer_architecture_by_version(version),
+                (None, None) => {
+                    return Err(GraphError::MissingParams(
+                        vec!["arch", "version"]
+                            .into_iter()
+                            .map(String::from)
+                            .collect(),
+                    ))
+                }
+            };
+
+            if !ARCH_VALIDATION_REGEX_RE.is_match(&arch) {
+                return Err(GraphError::InvalidParams(format!(
+                    "arch '{}' does not match regex '{}'",
+                    arch, ARCH_VALIDATION_REGEX_STR
+                )));
+            };
+
+            let mut graph = internal_io.graph;
+
+            // find all releases with a different the arch
+            let to_remove = {
+                graph
+                    .find_by_fn_mut(|release| {
+                        match release {
+                            crate::Release::Concrete(concrete_release) => concrete_release
+                                .metadata
+                                .get_mut(&format!("{}.{}", key_prefix, key_suffix))
+                                .map_or(true, |values| {
+                                    if values.split(',').any(|value| value.trim() == arch) {
+                                        *values = arch.clone();
+                                        false
+                                    } else {
+                                        true
+                                    }
+                                }),
+                            // remove if it's not a ConcreteRelease
+                            _ => true,
+                        }
+                    })
+                    .into_iter()
+                    .map(|(release_id, version)| {
+                        trace!("queuing '{}' for removal", version);
+                        release_id
+                    })
+                    .collect()
+            };
+
+            // remove all matches from the Graph
+            let removed = graph.remove_releases(to_remove);
+
+            trace!("removed {} releases", removed);
+
+            // remove the build suffix from the version
+            graph
+                .iter_releases_mut(|mut release| {
+                    let mut version = semver::Version::parse(release.version())?;
+                    version.build.retain(|elem| elem.to_string() != arch);
+                    let version = version.to_string();
+
+                    trace!("rewriting version {} ->  {}", release.version(), version);
+
+                    match &mut release {
+                        crate::Release::Abstract(release) => release.version = version,
+                        crate::Release::Concrete(release) => release.version = version,
+                    };
+
+                    Ok(())
+                })
+                .map_err(|e| GraphError::InternalError(e.to_string()))?;
```

Comment:
Let make it more specific - `GraphError::ArchVersionError`?

Reply:
Maybe `SemVerParseError`?